### PR TITLE
cmake style and some bugfixes

### PR DESCRIPTION
**Pull request by maage**
#### Description:

There is tool `cmake-lint` to ensure cmake files have common style. And then small number cmake related bug fixes.

#### Rationale:

Common code style simplifies later analyzing code. Both as lint style but also how dependencies are ordered.

`cmake/SSGCommon.cmake` has special `Folders should not be build inputs or outputs` but it was not followed.

#### Review Hints:

I guess `DERIVATIVE` issue should be checked, as when reading code old did not make sense.

Increase dependencies for obvious files. But it is not very broad. It is easy to add all ssg python files as deps when any python script is used and that can help in some use cases. But when ever there is templates and such it is of no use.

I'm not totally happy with `cmake-lint` comments and such.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -19,3 +19,7 @@ max_line_length = 99
 [{**.yml, **.yaml, **.jinja}]
 indent_style = space
 indent_size = 4
+
+[{CMakeLists.txt, **.cmake}]
+indent_style = space
+indent_size = 4
```
Comment (Mab879) [inline .editorconfig:25]: This project seems to use 2 spaces in CMake files, at least that was my experince when working with the CMake files.

**Review by Mab879 — COMMENTED** [review]
Thanks for the PR! We will need to expand our CMake style guide once this gets merged (or if you are willing, you can expand in this PR).

**Comment by maage** [comment]
Simplified patchset some, and rebased.

**Comment by marcusburghardt** [comment]
> Thanks for the PR! We will need to expand our CMake style guide once this gets merged (or if you are willing, you can expand in this PR).

@maage do you plan to update the Style Guide in this PR? It would be great.

**Comment by maage** [comment]
@marcusburghardt I could try to expand style guide in this PR.

**Comment by maage** [comment]
I reordered patches, style guide update and respective change. Now I changed also help files to conform stated style.

And the noticed a case where there was some `\$` when quotation does not seem to make sense.

**Review by marcusburghardt — APPROVED** [review]
Good job! It looks good to me, but I would prefer more reviews here since there are also some few changes beyond styles. @jan-cerny and @Mab879 , could you also take a look, please?

**Pull request merged**